### PR TITLE
Add 'original' (what the object looked like before this save) to the beforeSave hook

### PR DESCRIPTION
**Pull request by MatrixFrog**
The first argument to a beforeSave hook is a [TriggerRequest](http://parseplatform.org/Parse-SDK-JS/api/v1.11.1/Parse.Cloud.html#.TriggerRequest) which has a field called `original` with type `Parse.Object`

### CODE REVIEW

File: src/parse-mockdb.js [hunk 1]
```diff
@@ -687,9 +688,19 @@ function runHook(className, hookType, data) {
     const model = Parse.Object.fromJSON(modelJSON);
     hook = hook.bind(model);
 
+    const collection = getCollection(className);
+    const originalData = Object.assign({}, collection[model.id], { className });
+    let original;
+    if (originalData) {
+      const originalJSON = _.mapValues(originalData,
+        // Convert dates into JSON loadable representations
+        value => ((value instanceof Date) ? value.toJSON() : value)
```
Comment (esarachan) [inline src/parse-mockdb.js]: I'm not really familiar with this code at all, but it looks like in other places (like here: https://github.com/HustleInc/parse-mockdb/blob/master/src/parse-mockdb.js#L605) we're able to simply return the object as-is. Why do we need this extra logic here?

Comment (MatrixFrog) [reply src/parse-mockdb.js]: From what I could tell, what's stored in `collection[model.id]` is more or less the serialized form of the objects, while doing this and calling Parse.Object.fromJSON gives us a true Parse.Object, and in fact also gives us a subclass of Parse.Object, if we've defined one for the given className.

I'm still hoping to find official Parse docs that show we're doing this right. All I've found so far is https://docs.parseplatform.org/cloudcode/guide/#beforesave-triggers which doesn't mention the .original property at all.

Comment (MatrixFrog) [reply src/parse-mockdb.js]: Found it: The hook takes a TriggerRequest which is defined here http://parseplatform.org/Parse-SDK-JS/api/v1.11.1/Parse.Cloud.html#.TriggerRequest

**Comment by MatrixFrog** [comment]
Done. It doesn't end up saving that many lines but it's still better I guess.

**Pull request merged**